packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/classes.ts

```diff
@@ -86,22 +93,26 @@ export abstract class HandlerFrameworkClass extends ClassType {
    */
   public static buildFunction(scope: HandlerFrameworkModule, props: HandlerFrameworkClassProps): HandlerFrameworkClass {
     return new (class Function extends HandlerFrameworkClass {
-      protected readonly externalModules = [PATH_MODULE, CONSTRUCTS_MODULE, LAMBDA_MODULE];
-
       public constructor() {
         super(scope, {
           name: props.name,
           extends: LAMBDA_MODULE.Function,
           export: true,
         });
 
-        this.importExternalModulesInto(scope);
+        if (scope.isAlphaModule) {
+          scope.registerImport(LAMBDA_MODULE, { fromLocation: 'aws-cdk-lib/aws-lambda' });
```

Comment:
Correct. The import path changes based on where the file will be compiled. Since alpha modules have a dependency on aws-cdk-lib we need to use that as the import path.